src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunHashCode.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Text;
+
+using Internal.TypeSystem;
+
+namespace ILCompiler
+{
+    /// <summary>
+    /// Helper class used to calculate hash codes compatible with the CoreCLR
+    /// GetVersionResilientMethod/TypeHashCode.
+    /// </summary>
+    public static class ReadyToRunHashCode
+    {
+        /// <summary>
+        /// CoreCLR <a href="https://github.com/dotnet/coreclr/blob/030e0af89bb897554acef575075c69aaf5176268/src/vm/typehashingalgorithms.h#L15">ComputeNameHashCode</a>
+        /// </summary>
+        /// <param name="name">Name string to hash</param>
+        public static int NameHashCode(string name)
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                return 0;
+            }
+
+            int hash1 = 0x6DA3B944;
+            int hash2 = 0;
+
+            // DIFFERENT FROM CORERT: We hash UTF-8 bytes here, while CoreRT hashes UTF-16 characters.
```

Comment:
Ahh, of course, I completely missed this bit (most likely because Simon wrote this table). That definitely makes sense to put in sync with the new hashing code I ported over from CoreCLR. I tried to ask a couple of days back whether anyone's aware of existing managed code calculating the version resilient hashcodes but I didn't receive any response.

Reply:
Fixed in 6th commit.